src/gui/screen_lan_settings.c

```diff
@@ -73,11 +77,23 @@ static void _addrs_to_str(char *param_str, uint8_t flg) {
     strncpy(ip4_msk_str, ip4addr_ntoa(&(config.lan_ip4_msk)), IP4_ADDR_STR_SIZE);
     strncpy(ip4_gw_str, ip4addr_ntoa(&(config.lan_ip4_gw)), IP4_ADDR_STR_SIZE);
 
-    if (flg)
-        snprintf(param_str, MAX_INI_SIZE, "[lan_ip4]\naddress=%s\nmask=%s\ngateway=%s", ip4_addr_str, ip4_msk_str, ip4_gw_str);
-    else
-        snprintf(plan_str, 150, "IPv4 Address:\n    %s\nIPv4 Netmask:\n    %s\nIPv4 Gateway:\n    %s\nMAC Address:\n    %s",
+    if (flg){
+        char save_hostname[LAN_HOSTNAME_MAX_LEN + 1];
+        eeprom_get_hostname(save_hostname);
+        snprintf(param_str, MAX_INI_SIZE, "[lan_ip4]\ntype=%s\nhostname=%s\naddress=%s\nmask=%s\ngateway=%s",
+            config.lan_flag & LAN_EEFLG_TYPE ? LAN_type_opt[1] : LAN_type_opt[0], save_hostname, ip4_addr_str, ip4_msk_str, ip4_gw_str);
+    } else {
+        snprintf(plan_str, 150, "IPv4 Address:\n  %s      \nIPv4 Netmask:\n  %s      \nIPv4 Gateway:\n  %s      \nMAC Address:\n  %s",
```

Comment:
not sure about the spaces here, wouldn't a more precisely specified string placeholder work better? E.g. %-12.12s ?